Feature/rdb save

MoonSeongKwang opened the pull request:
### 변경점

1. 현재까지의 Storage Table의 변경사항을 다시 한 번 확인하여 반영하였습니다.
block_link와 register_code, 그리고 tsidx의 오타 등.

2. RDB에 pool session을 이용하여 자료가 저장되는것을 확인하였습니다.
문제가 되었던 특정 상황에서의 session 생성 시 Segmentation fault 문제는 WSL(Windows Subsystem for Linux) 환경에서의 에러일 가능성이 높아보입니다. 리눅스 서버에서 remote 작업으로 변경하였더니 잘 되고, Mac 환경에서도 문제가 없었습니다.
실제로 구동될 저장하는 코드는 추후 unresolved block pool의 과정에 들어가게 될 예정입니다.

MoonSeongKwang committed:
# EDIT - fisrt_block_test.json value change etc.
1. tx_id를 다시 계산하여 first_block_test.json을 수정하였습니다.
2. block을 구성하는 멤버 변수에 이전 블록의 sig를 갖는 link가 추가된
것을 반영하였습니다. 그로 인하여 block_id를 생성하는 validator의 관련
공식을 수정하였고, block.hpp를 수정하였습니다.

MoonSeongKwang committed:
# EDIT - RDB block info save, db structure update
RDB에 저장하는 간이 코드를 확인하였습니다.
db 구조가 바뀌어서 변경된 필드를 migrate하는 script에 반영하였습니다.
모든 rdb 저장 코드가 시험된것은 아니며, 현재 작성된 블록 저장은 정확하게
저장이 되고 있는지 확인이 필요합니다.

MoonSeongKwang committed:
# EDIT - Delete performQuery func
서로 다른 type의 변수를 저장해야하는 특성상 string형으로 통합하여
performQuery에서 일괄 처리하기가 어렵다고 판단하여 해당 함수를 삭제하고,
각 기능을 맡을 함수에서 개별적인 쿼리 처리를 할 예정입니다.

morishjs commented on src/plugins/chain_plugin/include/db_controller.hpp:
```diff
@@ -17,13 +18,12 @@ class DBController {
-  soci::connection_pool m_db_pool;
 
 public:
+  soci::connection_pool m_db_pool;
```
사소한거지만 현재 멤버변수에 대해 prefix로 m_을 안붙여도 됩니다. 그리고, 이렇게 직접 접근하게 두는 것보다 
```c++
soci::connection_pool& pool() {
  return m_db_pool; 
}
// db_controller.pool()
```
하는게 객체지향프로그래밍 관점에서 맞는 것 같습니다.

morishjs commented on src/plugins/chain_plugin/structure/block.hpp:
```diff
@@ -207,6 +211,10 @@ class Block {
     return m_transactions.size();
   }
 
+  string getAggz() {
```
string (x) -> base64_type(o)
참조: https://thevaulters.atlassian.net/wiki/spaces/SGN/pages/86638675/Block+Messages

MoonSeongKwang committed:
# FIX - Data input bug fix, etc.
 ### 변경사항
데이터가 정상적으로 저장되지 않고 앞 4바이트가 잘리던 문제를
수정했습니다.
soci 라이브러리에서 use()에 값을 넣을 때, string 값을 바로 넣으면 값
처리가 제대로 되지 않는 문제가 있어서 따로 string형 변수를 선언한 후
값을 할당해서 전달해야 정상적으로 동작합니다.

그리고 block.hpp의 get 함수에서 baseXX_type가 string으로 되어있던 곳들을
수정하였습니다.

SeongKwangMoon commented:
문자열의 값이 정확히 저장되고있지 않아서, 관련 버그를 수정하였습니다.

park53kr commented:
@SeongKwangMoon 
리뷰 다시 요청하고 싶으시면, dismiss 버튼 클릭 후 리뷰어 지정 다시 해주시면 됩니다. 그렇게 안하면 Notification이 안날아와서 다시 리뷰를 해야하는지 몰라요

SeongKwangMoon commented:
@park53kr 
넵 앞으로 그렇게 할게요
